compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1928,10 +1928,107 @@ OMR::ARM64::TreeEvaluator::BNDCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg
    return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
    }
 
+static TR::Instruction *compareIntsAndBranchForBNDCHK(TR::ARM64ConditionCode  branchType,
+                                           TR::Node             *node,
+                                           TR::CodeGenerator    *cg,
+                                           TR::SymbolReference  *sr)
+	{
+	TR::Node *secondChild = node->getSecondChild();
+	TR::Node *firstChild = node->getFirstChild();
+	TR::LabelSymbol *snippetLabel = generateLabelSymbol(cg);
+	TR::Register *src1Reg = cg->evaluate(firstChild);
+
+	bool foundConst = false;
+	if (secondChild->getOpCode().isLoadConst())
+		{
+		int64_t value = secondChild->get64bitIntegralValue();
+		if (constantIsUnsignedImm12(value))
+			{
+			generateCompareImmInstruction(cg, node, src1Reg, value, true);
+			foundConst = true;
+			}
+		else if (constantIsUnsignedImm12(-value))
+			{
+			generateCompareImmInstruction(cg, node, src1Reg, value, true);
+			foundConst = true;
+			}
+		}
+	if(!foundConst)
+		{
+		TR::Register *src2Reg = cg->evaluate(secondChild);
+		generateCompareInstruction(cg, node, src1Reg, src2Reg, true);
+		}
+
+	if (node->getNumChildren() == 3)
```

Comment:
Implementations of ArrayCopyBNDCHKEvaluator for other platforms don't seem to handle the case with a GlRegDeps child.